pkg/generate/app/componentresolvers.go

```diff
@@ -39,41 +41,36 @@ type PerfectMatchWeightedResolver []WeightedResolver
 // Resolve resolves the provided input and returns only exact matches
 func (r PerfectMatchWeightedResolver) Resolve(value string) (*ComponentMatch, error) {
 	imperfect := ScoredComponentMatches{}
-	var group WeightedResolvers
+	var group MultiSimpleSearcher
+	var groupWeight float32 = 0.0
 	for i, resolver := range r {
-		if len(group) == 0 || resolver.Weight == group[0].Weight {
-			group = append(group, resolver)
-			if i != len(r)-1 && r[i+1].Weight == group[0].Weight {
+		if len(group) == 0 || resolver.Weight == groupWeight {
+			group = append(group, resolver.Searcher)
+			groupWeight = resolver.Weight
+			if i != len(r)-1 && r[i+1].Weight == groupWeight {
 				continue
 			}
 		}
-		exact, inexact, err := resolveExact(group, value)
+		matches, err := group.Search(value)
 		switch {
-		case err != nil:
-			glog.V(5).Infof("Error from resolver: %v\n", err)
-			return nil, err
-		case exact != nil:
-			if exact.Score == 0.0 {
-				return exact, nil
-			}
-			if resolver.Weight != 0.0 {
-				exact.Score = resolver.Weight * exact.Score
+		case len(matches) > 0:
+			sort.Sort(ScoredComponentMatches(matches))
+			if matches[0].Score == 0.0 && (len(matches) == 1 || matches[1].Score != 0.0) {
+				return matches[0], nil
 			}
-			imperfect = append(imperfect, exact)
-		case len(inexact) > 0:
-			sort.Sort(ScoredComponentMatches(inexact))
-			if inexact[0].Score == 0.0 && (len(inexact) == 1 || inexact[1].Score != 0.0) {
-				return inexact[0], nil
-			}
-			for _, m := range inexact {
+			for _, m := range matches {
 				if resolver.Weight != 0.0 {
 					m.Score = resolver.Weight * m.Score
 				}
 				imperfect = append(imperfect, m)
```

Comment:
this name is confusing.  they are not necessarily imperfect matches, you could have a bunch of exact(perfect) matches in this array.